src/presets/cloudflare.ts

```diff
@@ -56,6 +67,46 @@ export const cloudflarePages = defineNitroPreset({
         resolve(nitro.options.output.serverDir, "path.js.map"),
         resolve(nitro.options.output.serverDir, "[[path]].js.map")
       );
+
+      const routes: CloudflarePagesRoutes = {
+        version: 1,
+        include: ["/*"],
+        exclude: [],
+      };
+
+      // Exclude public assets from hitting the worker
+      const explicitPublicAssets = nitro.options.publicAssets.filter(
+        (i) => !i.fallthrough
```

Comment:
User options as well as root path / check are applied in nitro.ts. `fallthrough` flag specified final behavior